### PR TITLE
PostgreSQL: add initial support (#4241) - v6

### PR DESCRIPTION
**Pull request by jufajardini**
Link to redmine ticket: https://redmine.openinfosecfoundation.org/issues/4241

Previous PR: #6630 

Describe changes:
- add config option to enable/disable password output (disabled by default)
- adjust parser to work with recent changes done in AppLayerTxData
- fix error that showed up in previous PR

Must still address:
- userguide documentation about pgsql (especially eve output)
- parse NoticeResponse & NotificationMessage, which could happen during any transaction
- decide on eve-log fields, since this is harder to change, once it's published (this includes logging info about `data_row` in a summarized way; probably hide up `password_message` altogether, if password output isn't enabled)
- understand and check GAP/ incomplete messages situation

suricata-verify-pr: 587
OISF/suricata-verify#587

### CODE REVIEW

File: rust/src/pgsql/parser.rs [hunk 1]
```diff
@@ -0,0 +1,2067 @@
...
+//         >> rows: dbg_dmp!(flat_map!(take!(length - 6), many_m_n!(0, field_count.into(), call!(parse_data_row_value))))
+//         >> (PgsqlBEMessage::DataRow(
+//             DataRowMessage {
+//                 identifier,
+//                 length,
+//                 field_count,
+//                 fields: rows,
+//             }
+//         ))
+//     ));
```
Comment (jufajardini) [inline rust/src/pgsql/parser.rs:846]: Delete this for next version, at least until we have extended logs.

File: rust/src/pgsql/parser.rs [hunk 2]
```diff
@@ -0,0 +1,2067 @@
...
+        b'E' => call!(pgsql_parse_error_response) |
+        b'K' => call!(parse_backend_key_data_message) |
+        b'N' => call!(pgsql_parse_notice_response) |
+        b'R' => call!(pgsql_parse_authentication_message) |
+        b'S' => call!(parse_parameter_status_message) |
+        b'C' => call!(parse_command_complete) |
+        b'Z' => call!(parse_ready_for_query) |
+        b'T' => call!(parse_row_description) |
+        b'D' => call!(dummy_parse_data_row)
+        // b'D' => call!(parse_data_row)
```
Comment (jufajardini) [inline rust/src/pgsql/parser.rs:977]: delete this.

File: rust/src/pgsql/parser.rs (hunk 1 above)
Comment (jufajardini) [reply rust/src/pgsql/parser.rs:846]: or altogether...

File: rust/src/pgsql/parser.rs [hunk 3]
```diff
@@ -0,0 +1,2067 @@
...
+            field_value: br#"28000"#.to_vec(),
+        };
+        let field3 = PgsqlErrorNoticeMessageField{
+            field_type: PgsqlErrorNoticeFieldType::Message,
+            field_value: br#"no pg_hba.conf entry for replication connection from host "192.168.50.11", user "rep", SSL off"#.to_vec(),
+        };
+        // let field4 = PgsqlErrorNoticeMessageField {
+        //     field_type: PgsqlErrorNoticeFieldType::TerminatorToken,
+        //     field_value: br#""#.to_vec(),
+        // };
```
Comment (jufajardini) [inline rust/src/pgsql/parser.rs:1359]: delete these

File: rust/src/pgsql/parser.rs [hunk 4]
```diff
@@ -0,0 +1,2067 @@
...
+
+        ok_res.push(field2);
+        ok_res.push(field3);
+
+        // let field4 = PgsqlErrorNoticeMessageField {
+        //     field_type: PgsqlErrorNoticeFieldType::TerminatorToken,
+        //     field_value: br#""#.to_vec(),
+        // };
+
+        // ok_res.push(field4);
```
Comment (jufajardini) [inline rust/src/pgsql/parser.rs:1378]: delete these and other commented out lines

**Comment by suricata-qa** [comment]
Information:
 
 | field | test | baseline | % |
|--|--|--|--| 
| | generic_stats_chk |
| .flow.spare | 1990044 | 1866612 | 106.61% | 

Pipeline 5065

**Comment by catenacyber** [comment]
Do you need a review of this Juliana ?
Or are you addressing the things you listed already first ?

**Comment by jufajardini** [comment]
> Do you need a review of this Juliana ? Or are you addressing the things you listed already first ?

I have addressed some, and am currently trying to move towards a final v1 that could be merged. As quite a few things have changed from this one, it's better to wait for the next one, I think. Thanks, @catenacyber ! :)

**Comment by jufajardini** [comment]
Followed by https://github.com/OISF/suricata/pull/6797

**Pull request closed without merging**